modules/ding_gatewayf/ding_gatewayf.module

```diff
@@ -241,6 +244,31 @@ function ding_gatewayf_callback() {
 }
 
 /**
+ * Make a salt based on request parameters.
+ *
+ * @param array $params
+ *   The values to extract from.
+ * @param string $ret
+ *   The generated salt to return.
+ *
+ * @return string
+ *   The salt.
+ */
+function ding_gatewayf_key_from_attributes(array $params, $ret = '') {
+  foreach ($params as $key => $value) {
+    /** @var array $value */
```

Comment:
Why is this needed? Given the subsequent if statement it should be clear that `$value` is not always an array.